src/pages/ReportSettingsPage.js

```diff
@@ -110,9 +111,17 @@ class ReportSettingsPage extends Component {
 
     render() {
         const shouldShowRoomName = !ReportUtils.isPolicyExpenseChat(this.props.report);
-        const shouldDisableRename = ReportUtils.isDefaultRoom(this.props.report) || ReportUtils.isArchivedRoom(this.props.report);
         const linkedWorkspace = _.find(this.props.policies, policy => policy && policy.id === this.props.report.policyID);
 
+        let shouldDisablePublicRoomRename = ReportUtils.isPublicRoom(this.props.report) && !linkedWorkspace;
+        if (ReportUtils.isPublicRoom(this.props.report) && linkedWorkspace) {
+            shouldDisablePublicRoomRename = !Policy.isPolicyOwner(linkedWorkspace) && linkedWorkspace.role !== CONST.POLICY.ROLE.ADMIN;
+        }
+
+        const shouldDisableRename = ReportUtils.isDefaultRoom(this.props.report)
+            || ReportUtils.isArchivedRoom(this.props.report)
+            || shouldDisablePublicRoomRename;
+
```

Comment:
NAB: I think if you put this in a function with early returns it may be more clear to see the different cases.